scripts/st2_bootstrap.sh

```diff
@@ -170,10 +170,15 @@ elif [[ -n "$DEBTEST" ]]; then
   echo "*** Detected Distro is ${DEBTEST} ***"
   SUBTYPE=`lsb_release -a 2>&1 | grep Codename | grep -v "LSB" | awk '{print $2}'`
   echo "*** Detected flavor ${SUBTYPE} ***"
-  if [[ "$SUBTYPE" != 'xenial' && "$SUBTYPE" != 'bionic' ]]; then
-    echo "Unsupported ubuntu flavor ${SUBTYPE}. Please use 16.04 (xenial) or 18.04 (bionic) as base system!"
-    exit 2
-  fi
+  case "$SUBTYPE" in
+    "xenial"|"bionic"|"focal")
+      true
+      ;;
+    *)
+      echo "Unsupported Ubuntu codename ${SUBTYPE}. Please use 16.04 (xenial), 18.04 (bionic) or 20.04 (focal) as base system!"
+      exit 2
+      ;;
+  esac
```

Comment:
Any specific reason for changing the construct here?

I'd prefer to keep the previous version, because it's less and easier code, but also aligns with the similar `EL` check above.

Reply:
I think `case` is easier to read and maintain for a single variable evaluation.

The `if` construct has the following advantages (+) / disadvantages (-):
 - (-) Repeats the `SUBTYPE` variable.
 - (+) Small code.
 - (-) Not easy to maintain (must apply logic and operator comparison) to add/remove a distro codename (SUBTYPE).
 - (-) Harder to read, codenames intermingled with logic/comparison operators.
 - (+) Well known construct. Same sort of logic as `if` in most other common languages.

The `case` construct has the following advantages (+) / disadvantages (-):
 - (+) `SUBTYPE` variable is used only once. DRY principle helps with maintainability.
 - (-) The code is larger.
 - (+) Easy to add/remove distro, just need to add the string, not manipulate the conditions.
 - (+) Easier to read. A single line displays the values and nothing else to match the codenames.
 - (-) Construct may not be well known.  People unfamiliar with shell scripting may not recognise this construct easily.

Reply:
That's a good breakdown. Looks indeed a bit verbose and I guess readability might be perceived differently.

Let's stay with the old variant for consistency with the EL checks in these scripts.
The main focus here is getting U20 ready.

Reply:
I can't see where `SUBTYPE` is used in the EL checks, can you point it out to me please?

Reply:
https://github.com/StackStorm/st2-packages/blob/9e84034d0f36b7c8ce829ed044f8af1c1dd92ec5/scripts/st2_bootstrap.sh#L162-L165

Reply:
Thanks @armab, I've reverted to the `if` construct.